src/utils/tradeMeaningFullyDiffer.ts

```diff
@@ -1,16 +1,16 @@
+import { Percent } from '@uniswap/sdk-core'
 import { InterfaceTrade } from 'state/routing/types'
 
 /**
  * Returns true if the trade requires a confirmation of details before we can submit it
  * @param args either a pair of V2 trades or a pair of V3 trades
  */
-export function tradeMeaningfullyDiffers(...args: [InterfaceTrade, InterfaceTrade]): boolean {
-  const [tradeA, tradeB] = args
+export function tradeMeaningfullyDiffers(tradeA: InterfaceTrade, tradeB: InterfaceTrade, slippage: Percent): boolean {
   return (
     tradeA.tradeType !== tradeB.tradeType ||
     !tradeA.inputAmount.currency.equals(tradeB.inputAmount.currency) ||
-    !tradeA.inputAmount.equalTo(tradeB.inputAmount) ||
+    tradeB.inputAmount.greaterThan(tradeA.maximumAmountIn(slippage)) ||
     !tradeA.outputAmount.currency.equals(tradeB.outputAmount.currency) ||
-    !tradeA.outputAmount.equalTo(tradeB.outputAmount)
+    tradeB.executionPrice.lessThan(tradeA.worstExecutionPrice(slippage))
```

Comment:
@tinaszheng could u confirm this is logic is accurate? in terms of which values I'm comparing

Reply:
yeah this logic seems fine to me! since `Price` is a ratio, you're technically capturing both the exact in/exact out differences in [one step](https://github.com/Uniswap/router-sdk/blob/main/src/entities/trade.ts#L228) which is clever! you then don't need the line above:
```
tradeB.inputAmount.greaterThan(tradeA.maximumAmountIn(slippage))
```

Reply:
ohhhhh ty